external/avro/src/main/scala/org/apache/spark/sql/avro/AvroDataToCatalyst.scala

```diff
@@ -53,7 +54,9 @@ case class AvroDataToCatalyst(
 
   @transient private lazy val avroSchema = new Schema.Parser().parse(jsonFormatSchema)
 
-  @transient private lazy val reader = new GenericDatumReader[Any](avroSchema)
+  @transient private lazy val reader = writerJsonFormatSchema
```

Comment:
So either writerJsonFormatSchema or jsonFormatSchema is used. In such case, if writerJsonFormatSchema is present, why not set writerJsonFormatSchema as jsonFormatSchema without changing the API?
Could you explain more how the API is called?

Reply:
No, if the writerJsonFormatSchema is set then both writerJsonFormatSchema and jsonFormatSchema are used (in the `.map`), otherwise only the jsonFormatSchema is used (in the `.getOrElse`).

In the first case the GenericDatumReader is built with the constructor accepting the two schemas (reader and writer) while in the second case it's built with the constructor accepting only one schema (which is assumed to be the same for both operations): https://avro.apache.org/docs/1.8.2/api/java/org/apache/avro/generic/GenericDatumReader.html

Please refer to the Jira ticket for a more detailed explanation: https://issues.apache.org/jira/browse/SPARK-27506

Reply:
I see. Sorry for the misunderstanding. 
I am now +1 for the proposal